<a href="https://colab.research.google.com/github/cksdlakstp12/Fire_Safety_AI_Competition/blob/main/FireAIPredictionScore.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import time
from datetime import datetime, timedelta
from calendar import monthrange

In [ ]:
df = pd.read_csv("grid_dataset.csv")

In [ ]:
# 모든 GRID_ID 중 출동 빈도가 가장 높은 GRID_ID순으로 출력
print(df.groupby("GRID_ID").sum().sort_values("DSP_CNT_IDEX", ascending=False)[["DSP_CNT_IDEX"]])

          DSP_CNT_IDEX
GRID_ID               
39545254          1085
39545274           961
39445274           708
39645264           608
39645244           511
...                ...
39145104             0
38645304             0
38645314             0
40245264             0
38745184             0

[687 rows x 1 columns]


In [ ]:
df.loc[df["GRID_ID"] == 39445274].to_csv("test.csv")

In [ ]:
inp_val = [] # 예측 점수를 평가할 입력값 리스트
d = datetime(2020, 1, 1) # 예측할 첫 날
lastd = datetime(2020, 12, 31) # 마지막 날

while d < lastd:
  _, lstday = monthrange(d.year, d.month) # lstday에는 해당 월의 마지막 날짜가 저장됨
  if d.day == lstday: # 마지막 날은 예측을 위해 스킵함
    d += timedelta(days=1)
    continue
  
  ymd = int(d.strftime("%Y%m%d")) # 해당 년월일에 대해,
  for t in range(0, 24): # 24시간동안,
    GRID_ID = 39545274 # 사고 발생 격자를 39545274(가장 사고 빈도가 높았던 GRID ID)로 선정
    inp_val.append((ymd, t, GRID_ID))
  
  d += timedelta(days=1)

In [ ]:
# inp_val = []
res = [] # 결과 리스트
for (ymd, t, grid_id) in inp_val: # YMD, TIME 상에 해당 GRID_ID의 DSP_CNT_IDEX가 1 이상인지를 확인하는 로직
  v = (df.loc[(df["YMD"] == ymd) & (df["TIME"] == t) & (df["GRID_ID"] == grid_id)]["DSP_CNT_IDEX"] >= 1).any()# 만약 특정 YMD, TIME, GRID_ID의 DSP_CNT_IDEX가 1 이상이면 1, 또는 0
  res.append(v) # 결과에 추가
print("예측 정확도:", np.mean(np.array(res))) # 결과의 평균 정확도를 출력

예측 정확도: 0.11240583804143126


GRID_ID 39545274

예측 정확도: 0.0981638418079096

GRID_ID 39545254

예측 정확도: 0.11240583804143126